packages/app/src/cli/services/versions-list.ts

```diff
@@ -86,6 +87,22 @@ async function getAppApiKey(token: string, options: VersionListOptions): Promise
   if (envIdentifiers.app) return envIdentifiers.app
```

Comment:
In my local case, `envIdentifiers.app` exists so the result returns before your check.

I think the approach here needs to be modified:

1. It doesn't work as is
2. It doesn't feel right that `getAppApiKey` is where the error pops up, when actually it should complete.

Instead, I'd recommend fetching the partners app in `versionList`, doing your check there, and then passing the result to `getAppApiKey`. That way, the check always runs, and the error is being raised by a function whose concern is actually at the level of versions listing.

Reply:
Makes sense, thanks!

Reply:
> is there no way to save the selected app after you've run the app versions list command once?

Yeah, I think if the partners app is fetched using `fetchAppAndIdentifiers()` (like in https://github.com/Shopify/cli/pull/2609/commits/4ea9e77f312329f17fb8d133c9408e3bd6871a9a), some data will be cached so that users won't have to go through the prompts to select their app each time they run the command. Can someone from @Shopify/cli-foundations confirm whether this is correct and review these changes?